src/read.c

```diff
@@ -2485,10 +2485,7 @@ avifResult avifDecoderReset(avifDecoder * decoder)
                 if (readResult != AVIF_RESULT_OK) {
                     return readResult;
                 }
-                if ((readData.data == NULL) || (readData.size != item->size)) {
-                    return AVIF_RESULT_BMFF_PARSE_FAILED;
-                }
-                if (!avifParseImageGridBox(&data->colorGrid, readData.data, item->size)) {
```

Comment:
At this point, we know readData.size == item->size, so we can replace item->size with readData.size, which looks nicer with readData.data.